ldap/servers/slapd/log.c

```diff
@@ -224,64 +217,41 @@ slapd_log_error_proc_internal(
-        rc = -1;                                                                                  \
-    }
-#define LOG_WRITE_NOW(fd, buffer, size, headersize, err)                                              \
-    do {                                                                                              \
-        (err) = 0;                                                                                    \
-        if (slapi_write_buffer((fd), (buffer), (PRInt32)(size)) != (PRInt32)(size)) {                 \
-            PRErrorCode prerr = PR_GetError();                                                        \
-            syslog(LOG_ERR, "Failed to write log, " SLAPI_COMPONENT_NAME_NSPR " error %d (%s): %s\n", \
-                   prerr, slapd_pr_strerror(prerr), (buffer) + (headersize));                         \
-            (err) = prerr;                                                                            \
-            rc = -1;                                                                                  \
-        }                                                                                             \
-        /* Should be a flush in here ?? Yes because PR_SYNC doesn't work ! */                         \
-        PR_Sync(fd);                                                                                  \
-    } while (0)
-#define LOG_WRITE_NOW_NO_ERR(fd, buffer, size, headersize)                                            \
-    do {                                                                                              \
-        if (slapi_write_buffer((fd), (buffer), (PRInt32)(size)) != (PRInt32)(size)) {                 \
-            PRErrorCode prerr = PR_GetError();                                                        \
-            syslog(LOG_ERR, "Failed to write log, " SLAPI_COMPONENT_NAME_NSPR " error %d (%s): %s\n", \
-                   prerr, slapd_pr_strerror(prerr), (buffer) + (headersize));                         \
-            rc = -1;                                                                                  \
-        }                                                                                             \
-        /* Should be a flush in here ?? Yes because PR_SYNC doesn't work ! */                         \
-        PR_Sync(fd);                                                                                  \
-    } while (0)
-#define LOG_CLOSE(fd) \
-    PR_Close((fd))
 
+static int32_t
+log_write(LOGFD fd, char *buffer, int32_t size, int32_t headersize)
+{
+    if (slapi_write_buffer((fd), (buffer), (PRInt32)(size)) != (PRInt32)(size)) {
+        PRErrorCode prerr = PR_GetError();
+        syslog(LOG_ERR, "Failed to write log, " SLAPI_COMPONENT_NAME_NSPR " error %d (%s): %s\n",
+               prerr, slapd_pr_strerror(prerr), (buffer) + (headersize));
+        return -1;
```

Comment:
Minor nitpick: For consistency, we can use the `rc` here, too (the same way it's used in `log_write_now`).

Reply:
> Minor nitpick: For consistency, we can use the `rc` here, too (the same way it's used in `log_write_now`).

Fixed.